### PR TITLE
Add get_origin_domain and get_shape to GridIndexing API

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Currently we use "add" arguments to get origin, domain, and shape values. This PR provides an interface on GridIndexing that allows us to get these values using the dimensionality of the data (i.e. the staggering information) and the number of halos required, which is clearer to understand.

## Code changes:

- Added get_origin_domain and get_shape to GridIndexing class
- Removed `origin` from GridIndexing init, since it can be derived from `n_halo`

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by mcgibbon** [comment]
launch jenkins

File: fv3core/utils/grid.py [hunk 1]
```diff
@@ -540,6 +544,54 @@ def domain_compute(self, add: Index3D = (0, 0, 0)):
...
+            dims: dimension names, using dimension constants from fv3gfs.util
+            halos: number of halo points for each dimension, defaults to zero
+
+        Returns:
+            origin: origin of the computation
+            domain: shape of the computation
+        """
+        origin = list(self._sizer.get_origin(dims))
+        domain = list(self._sizer.get_extent(dims))
+        print(origin, domain)
```
Comment (rheacangeo) [inline fv3core/utils/grid.py:564]: stray print statement

**Review by rheacangeo — APPROVED** [review]
Looks great!

File: fv3core/utils/grid.py (hunk 1 above)
Comment (mcgibbon) [inline fv3core/utils/grid.py:564]:
```suggestion
```

**Pull request merged**